### PR TITLE
feat: filter feature works

### PR DESCRIPTION
**Pull request by alswl**
Request https://github.com/Evertras/bubble-table/issues/5 is done.

demo `go run examples/filter/main.go`.

```
A filtered simple default table
Currently filter by:
Press q or ctrl+c to quit

┏━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━┓
┃        Title┃       Author┃Desc…┃
┣━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━┫
┃Computer Sys…┃Randal E. Br…┃This…┃
┃Effective Ja…┃ Joshua Bloch┃The …┃
┃Structure an…┃Harold Abels…┃Stru…┃
┃Game Program…┃Robert Nystr…┃The …┃
┣━━━━━━━━━━━━━┻━━━━━━━━━━━━━┻━━━━━┫
┃                              1/1┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

A filtered simple default table
Currently filter by:
Press q or ctrl+c to quit

┏━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━┓
┃        Title┃       Author┃Desc…┃
┣━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━┫
┃Computer Sys…┃Randal E. Br…┃This…┃
┃Structure an…┃Harold Abels…┃Stru…┃
┃Game Program…┃Robert Nystr…┃The …┃
┣━━━━━━━━━━━━━┻━━━━━━━━━━━━━┻━━━━━┫
┃                        /pro  1/1┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

┏━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━┓
┃        Title┃       Author┃Desc…┃
┣━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━┫
┃Game Program…┃Robert Nystr…┃The …┃
┣━━━━━━━━━━━━━┻━━━━━━━━━━━━━┻━━━━━┫
┃                       /game  1/1┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
```

### CODE REVIEW

File: table/column.go
Comment (Evertras) [inline table/column.go:13]: I would prefer not to export this, since `WithFiltered` then handles it when chained with `NewColumn`.

File: table/model.go
Comment (Evertras) [inline table/model.go:88]: Minor, but let's stick with fail fast pattern:

```golang
if column.Filter {
  continue
}
// ...
```

File: table/model.go
Comment (Evertras) [inline table/model.go:118]: I think we should rename this to `GetVisibleRows`, because `GetRows` implies it returns all the rows.

File: examples/filter/main.go
Comment (Evertras) [inline examples/filter/main.go:88]: Let's add something to say that `/` + typing will start filtering, and which column it will filter on.  When I first saw the example I thought it would filter on all.

**Review by Evertras — CHANGES_REQUESTED** [review]
Thanks for the submission!  A few things to look at.

I also want to consider how a user might supply filters/searches through the API.  For example, a user might have a text box outside the table to supply a filter term.  Can we add a way to supply the filter term as well as the text input you've added?  I do really like the `/` as I live in vim and that feels natural to me. :D

There will be room for expanding this in the future for things like numerical comparisons/filters, but I think that can definitely be another PR as long as we leave room for it.

File: go.mod
Comment (Evertras) [reply go.mod:13]: Seems fine, I see that happen sometimes too 🤔

File: table/model.go
Comment (alswl) [reply table/model.go]: Nice catch.

File: table/model.go
Comment (alswl) [reply table/model.go:85]: `GetVisibleRows` seems implies the rows that this page will shown. emm~ but it is better then `GetRows`.

Comment (Evertras) [reply table/model.go]: That's a very good point.  It doesn't take paging into account, but `GetVisibleRows` sounds like it should.  Hmm.

I'm open to ideas on this one, I'm not quite sure what to call it.  `GetSortedAndFilteredRows` feels a little silly, but... maybe?

Comment (alswl) [reply table/model.go]: I still using `GetVisibleRows` now.

Maybe `rows` for visible rows, and `allRows` for original rows. If that solution is approved, it can be another Pull Request.

**Comment by Evertras** [comment]
Just a note, we also need to make sure to fix the linting/testing coverage.  I unfortunately do want to stay strict with that, as it's caught many of my own mistakes even in this project.

**Comment by alswl** [comment]
> Just a note, we also need to make sure to fix the linting/testing coverage. I unfortunately do want to stay strict with that, as it's caught many of my own mistakes even in this project.

Sure.

**Comment by Evertras** [comment]
Hey there, just checking if you were able to add some coverage.  If not, I'm willing to help out here.  There appear to be some bugs regarding the selectable rows and another issue that would collide with the keymap added in this PR, so I'd like to unblock this if you find the test coverage requirements painful.

**Comment by alswl** [comment]
@Evertras I added as more unit tests as I can. And I merged with upstream main branch(solved some conflicts).

If the coverage is still under the required, help is wanted~

**Comment by Evertras** [comment]
Thanks!  Just a few linting things snuck in... a quick `make lint` locally should point them out as well.  Once those are fixed I'm happy to merge!

**Comment by alswl** [comment]
@Evertras thanks for point out. Style issues is fixed.

**Comment by Evertras** [comment]
Thank you for the patience, I'll take care of the coverage gap; I'll just not make a release until I fix that.  Appreciate the work!

**Comment by alswl** [comment]
@Evertras Maybe `availableRows` is better than `visibleRows`.

**Pull request merged**
